src/System.Device.Gpio/System/Device/Gpio/Drivers/SysFSDriver.Linux.cs

```diff
@@ -0,0 +1,562 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+
+namespace System.Device.Gpio.Drivers
+{
+    /// <summary>
+    /// A GPIO driver for Unix.
+    /// </summary>
+    internal class SysFSDriver : UnixDriver
+    {
+        private const string GpioBasePath = "/sys/class/gpio";
+        private int _pollFileDescriptor = -1;
+        private Thread _eventDetectionThread;
+        private int _pinsToDetectEventsCount;
+        private readonly static CancellationTokenSource s_eventThreadCancellationTokenSource = new CancellationTokenSource();
+        private readonly List<int> _exportedPins = new List<int>();
+        private readonly Dictionary<int, UnixDriverDevicePin> _devicePins = new Dictionary<int, UnixDriverDevicePin>();
+        private readonly int _pollingTimeoutInMilliseconds = Convert.ToInt32(TimeSpan.FromMilliseconds(1).TotalMilliseconds);
```

Comment:
Isn't this just 1? Why set it with all this ceremony?

Reply:
I this might be the existing code and the file just got renamed though :)

Reply:
yeah, its all just copied from UnixDriver, gonna ignore comments on this file